libcontainer/init_linux.go

```diff
@@ -516,7 +516,7 @@ func setupUser(config *initConfig) error {
 
 	// Before we change to the container's user make sure that the processes
 	// STDIO is correctly owned by the user that we are switching to.
-	if err := fixStdioPermissions(execUser); err != nil {
+	if err := fixStdioPermissions(execUser.Uid); err != nil {
```

Comment:
Do we still need this here if we have already done it in the parent?

Reply:
If there are different cases (in some we should do this from the parent, in some others from the child), maybe we should make calls to `fixStdioPermissions` conditional.

Reply:
> If there are different cases (in some we should do this from the parent, in some others from the child), maybe we should make calls to `fixStdioPermissions` conditional.

Since I considered that `Root in userNS != User in userNS`, so I think it still need in init process.
But after I add a test case for it, I find this is also a bug in userns container.
So I updated this PR to move `fixStdioPermissions` from init process to parent process.

Reply:
Thanks your comment to find another bug.

Reply:
Please see this test: @test "check stdio permission for non-root user in userns [terminal=false]"